From 86bc90fe03375a5e264eac935d5c9bb732e8108a Mon Sep 17 00:00:00 2001
From: Raghavender Reddy Bujala <quic_rbujala@quicinc.com>
Date: Thu, 13 Feb 2025 13:38:53 +0530
Subject: [PATCH] UPSTREAM: Audio Profiles version upgrade

Version changes
A2DP from 1.3 to 1.4
HFP from 1.7 to 1.8
AVRCP TG from 1.5 to 1.6
AVCTP from 1.3 to 1.4

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Git-commit: f0e9ecb9f4fe3a7bdb011dc155cf6ad657a289ee
Signed-off-by: Raghavender Reddy Bujala <quic_rbujala@quicinc.com>
Upstream-Status: Backport
---
 profiles/audio/a2dp.c  | 2 +-
 profiles/audio/avrcp.c | 6 +++---
 src/profile.c          | 4 ++--
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/profiles/audio/a2dp.c b/profiles/audio/a2dp.c
index b43161a..d8885eb 100644
--- a/profiles/audio/a2dp.c
+++ b/profiles/audio/a2dp.c
@@ -1581,7 +1581,7 @@ static sdp_record_t *a2dp_record(uint8_t type)
 	sdp_record_t *record;
 	sdp_data_t *psm, *version, *features;
 	uint16_t lp = AVDTP_UUID;
-	uint16_t a2dp_ver = 0x0103, avdtp_ver = 0x0103, feat = 0x000f;
+	uint16_t a2dp_ver = 0x0104, avdtp_ver = 0x0103, feat = 0x000f;
 
 	record = sdp_record_alloc();
 	if (!record)
diff --git a/profiles/audio/avrcp.c b/profiles/audio/avrcp.c
index dda9a30..def175f 100644
--- a/profiles/audio/avrcp.c
+++ b/profiles/audio/avrcp.c
@@ -131,7 +131,7 @@
 
 #define AVRCP_BROWSING_TIMEOUT		1
 #define AVRCP_CT_VERSION		0x0106
-#define AVRCP_TG_VERSION		0x0105
+#define AVRCP_TG_VERSION		0x0106
 
 #define AVRCP_SCOPE_MEDIA_PLAYER_LIST			0x00
 #define AVRCP_SCOPE_MEDIA_PLAYER_VFS			0x01
@@ -406,7 +406,7 @@ static sdp_record_t *avrcp_ct_record(bool browsing)
 	sdp_record_t *record;
 	sdp_data_t *psm[2], *version, *features;
 	uint16_t lp = AVCTP_CONTROL_PSM;
-	uint16_t avctp_ver = 0x0103;
+	uint16_t avctp_ver = 0x0104;
 	uint16_t feat = ( AVRCP_FEATURE_CATEGORY_1 |
 						AVRCP_FEATURE_CATEGORY_2 |
 						AVRCP_FEATURE_CATEGORY_3 |
@@ -482,7 +482,7 @@ static sdp_record_t *avrcp_tg_record(bool browsing)
 	sdp_record_t *record;
 	sdp_data_t *psm_control, *version, *features;
 	uint16_t lp = AVCTP_CONTROL_PSM;
-	uint16_t avctp_ver = 0x0103;
+	uint16_t avctp_ver = 0x0104;
 	uint16_t feat = ( AVRCP_FEATURE_CATEGORY_1 |
 					AVRCP_FEATURE_CATEGORY_2 |
 					AVRCP_FEATURE_CATEGORY_3 |
diff --git a/src/profile.c b/src/profile.c
index ea188f3..af03dbb 100644
--- a/src/profile.c
+++ b/src/profile.c
@@ -2069,7 +2069,7 @@ static struct default_settings {
 		.authorize	= true,
 		.auto_connect	= true,
 		.get_record	= get_hfp_hf_record,
-		.version	= 0x0107,
+		.version	= 0x0108,
 	}, {
 		.uuid		= HSP_HS_UUID,
 		.name		= "Headset unit",
@@ -2089,7 +2089,7 @@ static struct default_settings {
 		.authorize	= true,
 		.auto_connect	= true,
 		.get_record	= get_hfp_ag_record,
-		.version	= 0x0107,
+		.version	= 0x0108,
 		/* HFP 1.7.2: By default features bitfield is 0b001001 */
 		.features	= 0x09,
 	}, {
-- 
2.17.1

